torch/_prims/nvfuser_prims.py

```diff
@@ -199,6 +202,47 @@ def _{fname}_nvfuser(fd, a, b, c):
     )
 
 
+def _split_dim_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    a_shape: List[int],
+    dim: int,
+    outer_length: int,
+):
+    inner_length = a_shape[dim] // outer_length
+    new_shape = a_shape[0:dim] + [outer_length, inner_length] + a_shape[dim + 1 :]
+    return fd.ops.view(a, a_shape, new_shape)
+
+
+def _collapse_view_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    a_shape: List[int],
+    start: int,
+    end: int,
+):
+    dim_length = 1
+    for idx in range(start, end):
+        dim_length = dim_length * a_shape[idx]
+    new_shape = (
+        a_shape[0:start]
+        + [
+            dim_length,
+        ]
+        + a_shape[end:]
+    )
+    return fd.ops.view(a, a_shape, new_shape)
+
+
+def _reshape_nvfuser(
+    fd: Any,
+    a: TensorLikeType,
+    a_shape: List[int],
+    new_shape: List[int],
+):
+    return fd.ops.view(a, a_shape, new_shape)
```

Comment:
reshape isn't always `view-able`. I don't see a fallback/guard on this and it makes me nervous.

Reply:
Can we do a copy then a view just to be safe?

Reply:
NVM. this is just fusion record.....

Reply:
Is there a copy operation in nvfuser?
I think the "view" here is not the same as PyTorch's view, here we always create a copy as an output, right?

Reply:
Good catch~ Sorry I was being mindless. Our `view` is basically a reshape_copy. Please discard my comments here and below as well.

Fingers cross that the functionalize mutation from functorch would keep us safe. :crossed_fingers: